src/GitHubUtil.js

```diff
@@ -322,6 +322,23 @@ async function updateReference(ref, sha, force) {
     return (await rateLimitedPromise(promise)).object.sha;
 }
 
+async function updateReference(ref, sha, force) {
+    const max = 16 * 1000 + 1; // ~30 sec. overall
+    for (let d = 1000; d < max; d *= 2) {
+        try {
+            await updateReferenceRaw(ref, sha, force);
```

Comment:
Missing successful termination handling?

```suggestion
            return await updateReferenceRaw(ref, sha, force);
```